### PR TITLE
Dark mode CupertinoNavigationBar

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Update `CupertinoNavigationBar` and `CupertinoSliverNavigationBar` so they handle dynamic colors. No screenshots because the appearance change is minimal: https://github.com/flutter/goldens/pull/44

## Related Issues

part of https://github.com/flutter/flutter/issues/35541

## Tests

I added the following tests:

- goldens: https://github.com/flutter/goldens/pull/44
- Opaque background does not add blur effects, non-opaque background adds blur effects

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

Not sure if this is breaking as the golden changes seem to be minimal. 
- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 1]
```diff
@@ -427,14 +428,18 @@ class _CupertinoNavigationBarState extends State<CupertinoNavigationBar> {
       ),
     );
 
+    final Color actionsForegroundColor = CupertinoDynamicColor.resolve(
+      widget.actionsForegroundColor, // ignore: deprecated_member_use_from_same_package
+      context
```
Comment (justinmc) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: Trailing comma.

File: packages/flutter/test/cupertino/nav_bar_test.dart [hunk 2]
```diff
@@ -50,16 +51,35 @@ void main() {
...
+        theme: CupertinoThemeData(brightness: Brightness.light),
         home: CupertinoNavigationBar(
           middle: Text('Title'),
-          backgroundColor: Color(0xFFE5E5E5),
+          backgroundColor: background,
         ),
       ),
     );
     expect(find.byType(BackdropFilter), findsNothing);
+    expect(find.byType(CupertinoNavigationBar), paints..rect(color: const Color(0xFFE5E5E5)));
```
Comment (justinmc) [inline packages/flutter/test/cupertino/nav_bar_test.dart]: Could you use `background.color` for this color here?

File: packages/flutter/test/cupertino/nav_bar_test.dart [hunk 3]
```diff
@@ -50,16 +51,35 @@ void main() {
...
+      const CupertinoApp(
+        theme: CupertinoThemeData(brightness: Brightness.dark),
+        home: CupertinoNavigationBar(
+          middle: Text('Title'),
+          backgroundColor: background,
+        ),
+      ),
+    );
+    expect(find.byType(BackdropFilter), findsOneWidget);
+    expect(find.byType(CupertinoNavigationBar), paints..rect(color: const Color(0xF3E5E5E5)));
```
Comment (justinmc) [inline packages/flutter/test/cupertino/nav_bar_test.dart]: Also here, `background.darkColor`

**Review by justinmc — APPROVED** [review]
LGTM, just some small comments.

**Review by Piinks — APPROVED** [review]
LGTM, is this the minor golden change you mentioned offline? If so, I wouldn't consider it breaking. I would just make sure it is included in the change log.

**Pull request merged**